### PR TITLE
[DevCenter] Azure.Developer.DevCenter 1.0.0-beta.2

### PR DESCRIPTION
**Pull request by sebrenna**
This pull request includes changes to the Azure.Developer.DevCenter SDK that constitue version 1.0.0-beta.2. This SDK version is built around version **2022-11-11-preview** of our Data Plane API. The most notable change from 1.0.0-beta.1 is the use of an endpoint URI to construct clients, rather than tenant ID + dev center name, which was the architecture board's largest point of feedback for us in our initial review.

### CODE REVIEW

File: sdk/devcenter/Azure.Developer.DevCenter/CHANGELOG.md [hunk 1]
```diff
@@ -1,14 +1,7 @@
...
-## 1.0.0-beta.2 (Unreleased)
-
-### Features Added
-
-### Breaking Changes
-
-### Bugs Fixed
-
-### Other Changes
+## 1.0.0-beta.2 (2023-02-07)
```
Comment (pallavit) [inline sdk/devcenter/Azure.Developer.DevCenter/CHANGELOG.md:3]: Consider adding what changed between the 2 revisions.

Comment (sebrenna) [reply sdk/devcenter/Azure.Developer.DevCenter/CHANGELOG.md:3]: Added more details to the changelog.

Comment (pallavit) [reply sdk/devcenter/Azure.Developer.DevCenter/CHANGELOG.md:3]: Thank you!

**Comment by pallavit** [comment]
> var client = new DevCenterClient(

should this also be changed to endpoint?

---
Refers to: sdk/devcenter/Azure.Developer.DevCenter/README.md:42 in f6567a2. [](commit_id = f6567a214910e34b6d836ce6ddf631c06ca5e7a7, deletion_comment = False)

**Comment by pallavit** [comment]
>         internal HttpMessage CreateGetDevBoxesByUserNextPageRequest(string nextLink, string userId, string filter, int? maxCount, RequestContext context)

[nit] Consider consolidating common code in CreateXXXPageRequest methods into a single method. 

---
Refers to: sdk/devcenter/Azure.Developer.DevCenter/src/Generated/DevBoxesClient.cs:1199 in f6567a2. [](commit_id = f6567a214910e34b6d836ce6ddf631c06ca5e7a7, deletion_comment = False)

File: sdk/devcenter/Azure.Developer.DevCenter/tests/DevCenterClientTestEnvironment.cs [hunk 2]
```diff
@@ -20,7 +20,7 @@ public class DevCenterClientTestEnvironment : TestEnvironment
         private string _scope => GetRecordedVariable("DEFAULT_DEVCENTER_SCOPE");
         private string _testUserSecret => GetRecordedVariable("DEFAULT_TEST_USER_SECRET", options => options.IsSecret());
         private string _testUserName => GetRecordedVariable("DEFAULT_TEST_USER_NAME");
-        public string DevCenterName => GetRecordedVariable("DEFAULT_DEVCENTER_NAME");
+        public Uri Endpoint => new(GetRecordedVariable("DEVCENTER_ENDPOINT"));
```
Comment (pallavit) [inline sdk/devcenter/Azure.Developer.DevCenter/tests/DevCenterClientTestEnvironment.cs:23]: >DEVCENTER_ENDPOINT [](http://example.com/codeflow?start=56&length=18)

Curious why didn't you follow the same naming convention as the rest i.e. DEFAULT_DEVCENTER_ENDPOINT

Comment (sebrenna) [reply sdk/devcenter/Azure.Developer.DevCenter/tests/DevCenterClientTestEnvironment.cs:23]: Not sure. Will change it back

Comment (sebrenna) [reply sdk/devcenter/Azure.Developer.DevCenter/tests/DevCenterClientTestEnvironment.cs]: Switched back to the `DEFAULT_*` naming scheme!

**Review by ShivangiReja — APPROVED** [review]
LGTM

**Pull request merged**